lib/internal/http2/core.js

```diff
@@ -1887,45 +1762,44 @@ class ServerHttp2Stream extends Http2Stream {
 
     const streamOptions = options.endStream ? STREAM_OPTION_EMPTY_PAYLOAD : 0;
 
-    const ret = session[kHandle].submitPushPromise(this[kID],
-                                                   headersList,
-                                                   streamOptions);
+    const ret = this[kHandle].pushPromise(headersList, streamOptions);
     let err;
-    switch (ret) {
-      case NGHTTP2_ERR_NOMEM:
-        err = new errors.Error('ERR_OUTOFMEMORY');
-        process.nextTick(emit, session, 'error', err);
-        break;
-      case NGHTTP2_ERR_STREAM_ID_NOT_AVAILABLE:
-        err = new errors.Error('ERR_HTTP2_OUT_OF_STREAMS');
-        process.nextTick(emit, session, 'error', err);
-        break;
-      case NGHTTP2_ERR_STREAM_CLOSED:
-        err = new errors.Error('ERR_HTTP2_STREAM_CLOSED');
-        process.nextTick(emit, this, 'error', err);
-        break;
-      default:
-        if (ret <= 0) {
+    if (typeof ret === 'number') {
+      let target = session;
```

Comment:
It seems like we could just not set the `target` by default since each `case` branch redefines it, including `default`.

Or if we don't anticipate it changing, maybe just remove `target` altogether and replace it with hard-coded `this`. Can always change later if a session error comes to exist in this branch.

Reply:
+1